modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionProductPricer.java

```diff
@@ -126,6 +135,35 @@ public double impliedVolatility(
     }
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Provides the forward rate of the next fixing date
+   *
+   * @param swaption the swaption
+   * @param ratesProvider the rates provider
+   * @return the forward rate
+   */
+  public double forwardRate(ResolvedSwaption swaption, RatesProvider ratesProvider) {
+    RateIndex index = swaption.getIndex();
+    HolidayCalendar holidayCalendar = ReferenceData.standard().getValue(index.getFixingCalendar());
```

Comment:
This isn't right. there should never be a use of `ReferenceData.standard()`. Any reference data / jholiday calendar resolving needs to takes place earlier, with `ResolvedSwaption` containing the result. Same with `ZoneId.systemDefault()`. Normally, the `IborIndexObservation` would be in `ResolvedSwaption`.